os-server/src/main/java/app/openschool/course/CourseServiceImpl.java

```diff
@@ -1,6 +1,26 @@
 package app.openschool.course;
 
+import app.openschool.course.api.dto.CourseInfoDto;
+import app.openschool.course.api.mapper.CourseMapper;
+import java.util.Optional;
 import org.springframework.stereotype.Service;
 
 @Service
-public class CourseServiceImpl implements CourseService {}
+public class CourseServiceImpl implements CourseService {
+
+  private final CourseRepository courseRepository;
+
+  public CourseServiceImpl(CourseRepository courseRepository) {
+    this.courseRepository = courseRepository;
+  }
+
+  @Override
+  public Optional<CourseInfoDto> findCourseById(Long id) {
+    Optional<Course> optionalCourse = courseRepository.findById(id);
+    Optional<CourseInfoDto> optionalCourseInfoDto = Optional.empty();
+    if (optionalCourse.isPresent()) {
+      optionalCourseInfoDto = Optional.of(CourseMapper.toCourseInfoDto(optionalCourse.get()));
+    }
+    return optionalCourseInfoDto;
```

Comment:
return courseRepository.findById(id).map(CourseMapper::toCourseInfoDto).orElse(Optional.empty()); will do the trick I guess